File: hw/tipan/u2/hw16Libreries/JavaScript/model/product.js
const Tax = require("../utils/tax");

class Product {

    constructor(id, description, price, pvp = null) {

        this.id = id;
        this.description = description;
        this.price = price;

        if (pvp === null) {
            this.pvp = Tax.computeTotal(price, 15);
        } else {
            this.pvp = pvp;
        }
    }

    toString() {

        return `Product(id=${this.id}, description=${this.description}, price=${this.price}, pvp=${this.pvp})`;
    }
}

module.exports = Product;

File: hw/basurto/u2/hw09Libraries/Python/StoreApp/product.py
from taxlib import compute_total


class Product:

    def __init__(self, id, description, price):
        self.id = id
        self.description = description
        self.price = price

        self.pvp = compute_total(price, 15)

    def __str__(self):
        return f"""
Product
{{
    id={self.id}
    description={self.description}
    price={self.price}
    pvp={self.pvp}
}}
"""